src/state/state.js

```diff
@@ -188,14 +188,14 @@
          * @ignore
          */
         function _renderFrame(time) {
-            // update all game objects
-            me.game.update(time);
-            // render all game objects
-            me.game.draw();
             // schedule the next frame update
             if (_animFrameId !== -1) {
                 _animFrameId = window.requestAnimationFrame(_renderFrame);
             }
+            // update all game objects
+            me.game.update(time);
+            // render all game objects
+            me.game.draw();
```

Comment:
I get that, but requestAnimationFrame is an asynchronous callback. Once all of the single-threaded code for one execution of the loop is done, there's a period of time where the code is either waiting, doing nothing, for the callback to come *or* the callback is already resolved and executes immediately after the other synchronous code. So there's a variable here of how long it takes for the asynchronous callback to come. That's the logic, in theory.